language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/TypeCastCodeAction.java

```diff
@@ -114,7 +114,18 @@
                 || !isNumeric(rhsTypeSymbol.get()))) {
             return Collections.emptyList();
         }
-        String typeName = lhsTypeSymbol.get().signature();
+
+        String typeName;
+        if (lhsTypeSymbol.get().typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            Optional<String> name = lhsTypeSymbol.get().getName();
+            if (name.isEmpty()) {
+                return Collections.emptyList();
+            }
+            typeName = name.get();
+        } else {
+            typeName = lhsTypeSymbol.get().signature();
+        }
+        
```

Comment:
Shall we use `CommonUtil.getModifiedTypeName`, utility API?

Reply:
Done